src/module_2/module_2_data.md

```diff
@@ -0,0 +1,1750 @@
+
+# 2. Joining dataframes
+
+## 2.1. Regulars and users
+
+
+```python
+num_regulars = df_regulars.groupby('user_id')['variant_id'].nunique().reset_index().rename(columns={'variant_id':'num_regulars'})
+df_users_orders = df_users.merge(num_regulars, how='left', on='user_id').fillna({'num_regulars':0})
+display(df_users_orders.head())
+```
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>user_id</th>
+      <th>user_segment</th>
+      <th>user_nuts1</th>
+      <th>first_ordered_at</th>
+      <th>customer_cohort_month</th>
+      <th>count_people</th>
+      <th>count_adults</th>
+      <th>count_children</th>
+      <th>count_babies</th>
+      <th>count_pets</th>
+      <th>num_regulars</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0</th>
+      <td>0e823a42e107461379e5b5613b7aa00537a72e1b0eaa7a...</td>
+      <td>Top Up</td>
+      <td>UKH</td>
+      <td>2021-05-08 13:33:49</td>
+      <td>2021-05-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>0.0</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>15768ced9bed648f745a7aa566a8895f7a73b9a47c1d4f...</td>
+      <td>Top Up</td>
+      <td>UKJ</td>
+      <td>2021-11-17 16:30:20</td>
+      <td>2021-11-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>0.0</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>33e0cb6eacea0775e34adbaa2c1dec16b9d6484e6b9324...</td>
+      <td>Top Up</td>
+      <td>UKD</td>
+      <td>2022-03-09 23:12:25</td>
+      <td>2022-03-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>0.0</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>57ca7591dc79825df0cecc4836a58e6062454555c86c35...</td>
+      <td>Top Up</td>
+      <td>UKI</td>
+      <td>2021-04-23 16:29:02</td>
+      <td>2021-04-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>0.0</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>085d8e598139ce6fc9f75d9de97960fa9e1457b409ec00...</td>
+      <td>Top Up</td>
+      <td>UKJ</td>
+      <td>2021-11-02 13:50:06</td>
+      <td>2021-11-01 00:00:00</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>NaN</td>
+      <td>0.0</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+```python
+#Let's check the distribution of the number of regular products per user
+df_users_orders['num_regulars'].hist(bins=50, grid=False, color='#86bf91', log=True)
```

Comment:
What do you want to get from this? Also, how much does this correlate with the number of order per user?

Reply:
First I'd like to see visually the distribution of the number of regular items. I haven't checked the correlation with the number of orders per user. If checked, we could conclude that maybe people with more orders (more loyal customers) have a higher number of regular items.